src/NuGet.Services.Publish/PowerShellHashtableVisitor.cs

```diff
@@ -0,0 +1,157 @@
+        public object VisitForStatement(ForStatementAst forStatementAst) { throw new UnexpectedElementException(); }
+        public object VisitWhileStatement(WhileStatementAst whileStatementAst) { throw new UnexpectedElementException(); }
+        public object VisitCatchClause(CatchClauseAst catchClauseAst) { throw new UnexpectedElementException(); }
+        public object VisitTryStatement(TryStatementAst tryStatementAst) { throw new UnexpectedElementException(); }
+        public object VisitBreakStatement(BreakStatementAst breakStatementAst) { throw new UnexpectedElementException(); }
+        public object VisitContinueStatement(ContinueStatementAst continueStatementAst) { throw new UnexpectedElementException(); }
+        public object VisitReturnStatement(ReturnStatementAst returnStatementAst) { throw new UnexpectedElementException(); }
+        public object VisitExitStatement(ExitStatementAst exitStatementAst) { throw new UnexpectedElementException(); }
+        public object VisitThrowStatement(ThrowStatementAst throwStatementAst) { throw new UnexpectedElementException(); }
+        public object VisitDoUntilStatement(DoUntilStatementAst doUntilStatementAst) { throw new UnexpectedElementException(); }
+        public object VisitAssignmentStatement(AssignmentStatementAst assignmentStatementAst) { throw new UnexpectedElementException(); }
+        public object VisitCommand(CommandAst commandAst) { throw new UnexpectedElementException(); }
+        public object VisitCommandExpression(CommandExpressionAst commandExpressionAst) { throw new UnexpectedElementException(); }
+        public object VisitCommandParameter(CommandParameterAst commandParameterAst) { throw new UnexpectedElementException(); }
+        public object VisitFileRedirection(FileRedirectionAst fileRedirectionAst) { throw new UnexpectedElementException(); }
+        public object VisitMergingRedirection(MergingRedirectionAst mergingRedirectionAst) { throw new UnexpectedElementException(); }
+        public object VisitExpandableStringExpression(ExpandableStringExpressionAst expandableStringExpressionAst) { throw new UnexpectedElementException(); }
+        public object VisitIndexExpression(IndexExpressionAst indexExpressionAst) { throw new UnexpectedElementException(); }
+        public object VisitAttributedExpression(AttributedExpressionAst attributedExpressionAst) { throw new UnexpectedElementException(); }
+        public object VisitBlockStatement(BlockStatementAst blockStatementAst) { throw new UnexpectedElementException(); }
+        public object VisitInvokeMemberExpression(InvokeMemberExpressionAst invokeMemberExpressionAst) { throw new UnexpectedElementException(); }
+
+        public object VisitStatementBlock(StatementBlockAst statementBlockAst)
+        {
+            return statementBlockAst?.Statements?.FirstOrDefault()?.Visit(this);
```

Comment:
Are you using the newest compiler? Null propagation operator is a new C# 6 language feature.